tempodb/encoding/vparquet/block_traceql.go

```diff
@@ -575,29 +593,41 @@ func createStringPredicate(op traceql.Operator, operands traceql.Operands) (parq
 		return nil, nil
 	}
 
-	vals := make([]string, 0, len(operands))
-
 	for _, op := range operands {
 		if op.Type != traceql.TypeString {
 			return nil, fmt.Errorf("operand is not string: %+v", op)
 		}
-		vals = append(vals, op.S)
 	}
 
+	s := operands[0].S
```

Comment:
Logic here and above is a bit brittle. Why validate operands if only one is taken? Shouldn't the length be checked here not to panic?

Reply:
Agree, these parts are a little unclear.  All operations have a slice of operands, but each operation expects a different number.  The number of operands is checked for correctness earlier in this file in [checkConditions](https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet/block_traceql.go#L108)  In the following block, all operations take the one argument, and it is guaranteed to exist since the count was checked already.  OpNone exits early at the top.